vllm/model_executor/parameter.py

```diff
@@ -0,0 +1,191 @@
+from typing import Callable, List, Optional, Tuple, Union
+
+import torch
+from torch.nn import Parameter
+
+from vllm.distributed import get_tensor_model_parallel_rank
+from vllm.logger import init_logger
+
+__all__ = [
+    "vLLMParameter", "PackedvLLMParameter", "PerTensorScaleParameter",
+    "ModelWeightParameter", "ChannelQuantScaleParameter",
+    "GroupQuantScaleParameter"
+]
+
+logger = init_logger(__name__)
+
+
+class vLLMParameter(Parameter):
+
+    def __new__(cls, data: torch.Tensor, **kwargs):
+
+        return super().__new__(cls, data=data, requires_grad=False)
+
+    def __init__(self,
+                 data: torch.Tensor,
+                 weight_loader: Callable,
+                 ignore_warnings: Optional[bool] = True):
+
+        self._ignore_warnings = True
+        self._weight_loader = weight_loader
+
+    @property
+    def weight_loader(self):
+        return self._weight_loader
+
+
+class _ColumnvLLMParameter(vLLMParameter):
+
+    def __init__(self, output_dim: int, **kwargs):
+        self._output_dim = output_dim
+        super().__init__(**kwargs)
+
+    @property
+    def output_dim(self):
+        return self._output_dim
+
+    def load_column_parallel_weight(self, loaded_weight: torch.Tensor,
```

Comment:
These functions do not `load` the weights but rather `reshape`

We should move the 

```python
assert param_data.shape == loaded_weight.shape
param_data.copy_(loaded_weight)
```

Into this

Reply:
No because we use this assertion and copy for everything else that doesnt require sharding or param/loaded weight manipulation 

Such as our input scales for activation quantization

Reply:
We can update the names of the parameter functions to use a better verb than `load`

Reply:
But input scales will be `PerTensorScale` no?

Isn't the idea that all parameters will call `load_xxx_`?

Reply:
No it won't be
We also have other parameters that dont require the manipulation - such as the weight_shape for packed cases

Reply:
Sure, and `BasevLLMParameter` will have the passthrough right?

Also - why doesn't InputScale have the same manipulation? We still have 3 scales but the cutlass kernels only accept 1 scale no?